src/RollImage.cpp

```diff
@@ -3261,9 +3262,19 @@ void RollImage::recalculateFirstMusicHole(void) {
 			row = ta[i][j]->origin.first;
 			if (row < minrow) {
 				minrow = row;
+			} else if (row < nextMusicRow) {
+				nextMusicRow = row;
 			}
 		}
 	}
+
+	// Use a conservative distance (2.8 ft) so that only punctures and obvious
+	// outliers appearing well above the rest of the roll are disregarded.
+	if ((nextMusicRow > minrow) && ((nextMusicRow - minrow) > 10000)) {
```

Comment:
Not requesting a change, but just noting that this value works out to > 20 seconds even at 8ft/min, right?  This value is anecdotally validated of course (de facto, since we're basically using this in production already) -- but might there be merit in experimenting with something less conservative?  Or have we already got all the juice we can use out of this?